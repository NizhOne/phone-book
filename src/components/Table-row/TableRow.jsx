import React from 'react';
import PropTypes from 'prop-types';

import './table-row.css';

const TableRow = ({rowData, isHeader, deleteItem, editItem}) => {
    const buttons = isHeader ?
        null :
        <div>
            <button className={`table-row__button`} onClick={ (e, id) => {e.preventDefault(); deleteItem(id)} }>delete</button>
            <button className={`table-row__button`} onClick={ (e, id) => {e.preventDefault(); editItem(id)} }>edit</button>
        </div>;
    return (
        <div className={`table-row`}>
            <div className={`table-row__determining`}>
                {buttons}
                <span className={`table-row__determining_main`}>{rowData.name}</span>&nbsp;
            </div>
            <div className={`table-row__values`}>
                <div className={`table-row__values_value`}>{rowData.phoneNumber}</div>
                <div className={`table-row__values_value`}>{rowData.email}</div>
            </div>
        </div>
    );
};

TableRow.propTypes = {
    rowData: PropTypes.object.isRequired,
    isHeader: PropTypes.bool,
    deleteItem: PropTypes.func,
    editItem: PropTypes.func
};

export default TableRow