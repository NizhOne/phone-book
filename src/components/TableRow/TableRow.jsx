import React from 'react';
import PropTypes from 'prop-types';

import './table-row.css';

const TableRow = ({rowData, isHeader, deleteItem, editItem}) =>
    <div className={`table-row`}>
        <div className={`table-row__determining`}>
            {!isHeader &&
            <div>
                <button className={`table-row__button`} onClick={(e, id) => {
                    e.preventDefault();
                    deleteItem(id);
                }}>
                    delete
                </button>
                <button
                    className={`table-row__button`}
                    onClick={(e, id) => {
                        e.preventDefault();
                        editItem(id);
                    }}
                >
                    edit
                </button>
            </div>
            }
            <span className={`table-row__determining_main`}>{rowData.name}</span>&nbsp;
        </div>
        <div className={`table-row__values`}>
            <div className={`table-row__values_value`}>{rowData.phoneNumber}</div>
            <div className={`table-row__values_value`}>{rowData.email}</div>
        </div>
    </div>;

TableRow.propTypes = {
    rowData: PropTypes.object.isRequired,
    isHeader: PropTypes.bool,
    deleteItem: PropTypes.func,
    editItem: PropTypes.func
};

export default TableRow;