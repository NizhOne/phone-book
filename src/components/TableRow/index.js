export { default } from './Table-row'