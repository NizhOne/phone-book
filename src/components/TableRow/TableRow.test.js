import React from 'react';
import TableRow from './TableRow';
import renderer from 'react-test-renderer';


test('test', () => {
    const component = renderer.create(
        <TableRow rowData={{}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});