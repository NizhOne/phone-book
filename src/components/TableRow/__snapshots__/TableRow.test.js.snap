// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
<div
  className="table-row"
>
  <div
    className="table-row__determining"
  >
    <div>
      <button
        className="table-row__button"
        onClick={[Function]}
      >
        delete
      </button>
      <button
        className="table-row__button"
        onClick={[Function]}
      >
        edit
      </button>
    </div>
    <span
      className="table-row__determining_main"
    />
    Â 
  </div>
  <div
    className="table-row__values"
  >
    <div
      className="table-row__values_value"
    />
    <div
      className="table-row__values_value"
    />
  </div>
</div>
`;
